graph-deps

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+#
+# This script takes a dmd dependency file, and outputs a graph in graphviz
+# format also, detect cycles
+#
+# Based on: https://github.com/dellort/lumbricus/blob/d2/src/scripts/deps.py
+# Which used code from: http://www.logarithmic.net/pfh/blog/01208083168
+# (see https://github.com/sociomantic-tsunami/makd/pull/19 for details on
+# license changes)
+#
+# Copyright 2009-2016 Vincent Lang.
+# Copyright 2016 Sociomantic Labs GmbH.
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+from optparse import OptionParser
+import sys
+import re
+
+std_ignore = ["object", "tango.", "std.", "core."]
+
+def print_help(option, opt, value, parser, do_exit=True):
+    parser.print_help()
+    print ""
+    print "Generate the dependency file with:"
+    print "    dmd -o- -deps=depfile rootfile.d"
+    print ""
+    print "Example to generate a graph with circular dependencies only:"
+    print "    %s -c -C depfile graph.out" % parser.get_prog_name()
+    print ""
+    print "Render graph.out with:"
+    print "    dot -T svg graph.out -o graph.svg"
+    if do_exit: sys.exit(1)
+
+parser = OptionParser(usage="%prog [options] depfile graph.out",
+    description="Create a graphviz dependency graph from dmd module dependency file.",
+    add_help_option=False)
+parser.add_option("-h", "--help",
+    action="callback", callback=print_help,
+    help="show this help message and exit")
+parser.add_option("-c", "--cycles-only",
+    action="store_true", dest="cycles_only", default=False,
+    help="show only nodes, that are parts of a cycle")
+parser.add_option("-C", "--cycle-edges-only",
+    action="store_true", dest="cycle_edges_only", default=False,
+    help="show only edges, that are part of a cycle")
+parser.add_option("-e", "--exclude",
+    action="append", dest="ignore", default=[],
+    help="ignore module (or if it ends with a '.', ignore all packages starting with this)")
```

Comment:
maybe? `(or ignore all packages starting with this value, if it ends with a `.`)`